cpp/src/com/google/idea/blaze/cpp/CppSupportChecker.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2024 The Bazel Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.idea.blaze.cpp;
+
+import com.google.common.collect.ImmutableList;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.extensions.ExtensionPointName;
+import java.nio.file.Path;
+import java.util.List;
+
+/** Checks whether or not the IDE supports a given CPP configuration. */
+public interface CppSupportChecker {
+
+  ExtensionPointName<CppSupportChecker> EP_NAME =
+      new ExtensionPointName<>("com.google.idea.blaze.cpp.CppSupportChecker");
+
+  Logger logger = Logger.getInstance(CppSupportChecker.class);
+
+  static boolean isSupportedCppConfiguration(List<String> compilerSwitches, Path workspaceRoot) {
```

Comment:
Have 2 comments to this implementation

1. Basically it's a `return A || B || C || ...` what looks slightly loose to me. I guess if at least 1 checker disallows then it should return false. So it should be `A && B && C`. 
2. Performance note: to get `true` for clwb we copy list of parameters here and do some extra logic to get compiler flags in workspace/qsync code. And it happens for every resolve configuration. I would consider changing the interface from the current one into something like

```
boolean requiresAdditionalArgumentsCheck(); 
boolean isSupportedCppConfiguration(...);
```

or maybe the check can be incapsulated with the something like

```
isSupportedCppConfiguration(CidrCompilerSwitches switches)
```

Reply:
@mathewinwood as the author of the original commit. Can you address these comments?

Reply:
> Basically it's a return `A || B || C || ... ` 
True, but I don't think it would make any difference since currently there's at most 1 implementation per IDE configuration.

>  for clwb we copy list of parameters here and do some extra logic to get compiler flags 

Fair enough, I can change this.

Reply:
> True, but I don't think it would make any difference since currently there's at most 1 implementation per IDE configuration.

This leaves a lot of space to abuse, it's so easy to add a new extension which just overrides NDK checker. I'd prefer to see `AND` there. 

> Fair enough, I can change this.

Thank you

Reply:
Ok, I see. I can update it to be an AND .

Reply:
Not sure if we on the same page here, though I'm expecting the change that changes OR to AND. Is my expectation correct?

Reply:
That change is still in progress (sorry for the delay).